hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -1473,10 +1474,7 @@ public Properties build() {
       tableConfig.setValue(HoodieTableConfig.NAME, tableName);
       tableConfig.setValue(HoodieTableConfig.TYPE, tableType.name());
 
-      if (null == tableVersion) {
-        tableVersion = HoodieTableVersion.current();
-      }
-
+      tableVersion = getTableVersionProperly();
```

Comment:
This change might not be needed, because the `hoodie.write.table.version` only allows 6, 8, and 9 as valid values.  Could you try reverting the change and tests in `TestHoodieTableMetaClient` and see if the new tests in `TestUpgradeDowngrade` still pass?